### PR TITLE
make-bm-worker: add some options and fix some issues

### PR DESCRIPTION
**Pull request by zhouhao3**
* Improve the options of make-bm-worke so that it can generate more detailed yaml
* fix the position of disableCertificateVerification
* add some UT
* add the effective value of boot-mode argument
* fix consumer-namespace in template

### CODE REVIEW

**Comment by hardys** [comment]
> Regular contributors should join the org to skip this step.

Hi @zhouhao3 - thanks for your PR, I can see you've been regularly contributing so I sent an invite to join the org, this means your PRs will not require the manual ok-to-test step in future :)

**Comment by zhouhao3** [comment]
@fmuyassarov @andfasano @s3rj1k @dtantsur PTAL

File: cmd/make-bm-worker/main.go [hunk 1]
```diff
@@ -48,6 +56,25 @@ func main() {
 		os.Exit(1)
 	}
 
+	if *automatedCleaningMode != "" && *automatedCleaningMode != "metadata" && *automatedCleaningMode != "disabled" {
+		fmt.Fprintf(os.Stderr, "Invalid automatic cleaning mode %q, use \"metadata\" or \"disabled\"\n", *bootMode)
```
Comment (fmuyassarov) [inline cmd/make-bm-worker/main.go:60]:
```suggestion
		fmt.Fprintf(os.Stderr, "Invalid automatic cleaning mode %q, use \"metadata\" or \"disabled\"\n", *automatedCleaningMode)
```

File: cmd/make-bm-worker/main.go [hunk 2]
```diff
@@ -22,6 +22,14 @@ func main() {
 		"consumer", "", "specify name of a related, existing, consumer to link")
 	var consumerNamespace = flag.String(
 		"consumer-namespace", "", "specify namespace of a related, existing, consumer to link")
+	var automatedCleaningMode = flag.String(
+		"automatedCleaningMode", "", "automatic cleaning mode for host (metadata or disabled)")
+	var imageURL = flag.String("image-url", "", "url for the image")
+	var imageChecksum = flag.String("image-checksum", "", "checksum for the image")
+	var imageChecksumType = flag.String(
+		"image-checksum-type", "", "checksum algorithm for the image (md5, sha256 or sha256)")
```
Comment (furkatgofurov7) [inline cmd/make-bm-worker/main.go:30]: duplicate `sha256`, should be `sha512`

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Comment by fmuyassarov** [comment]
@zhouhao3 this looks good to me. Can we squash the commits before merging ?

**Comment by zhouhao3** [comment]
> Can we squash the commits before merging ?

Done, PTAL.

**Review by fmuyassarov — COMMENTED** [review]
Thank you!
/lgtm

**Pull request merged**